cmd/routed-eni-cni-plugin/driver/driver.go

```diff
@@ -267,354 +271,358 @@ func WaitForAddressesToBeStable(ifName string, timeout time.Duration) error {
 	}
 }
 
-// SetupNS wires up linux networking for a pod's network
-func (os *linuxNetwork) SetupNS(hostVethName string, contVethName string, netnsPath string, v4Addr *net.IPNet, v6Addr *net.IPNet,
-	deviceNumber int, vpcCIDRs []string, useExternalSNAT bool, mtu int, log logger.Logger) error {
-	log.Debugf("SetupNS: hostVethName=%s, contVethName=%s, netnsPath=%s, deviceNumber=%d, mtu=%d", hostVethName, contVethName, netnsPath, deviceNumber, mtu)
-	return setupNS(hostVethName, contVethName, netnsPath, v4Addr, v6Addr, deviceNumber, vpcCIDRs, useExternalSNAT, os.netLink, os.ns, mtu, log, os.procSys)
-}
-
-func setupNS(hostVethName string, contVethName string, netnsPath string, v4Addr *net.IPNet, v6Addr *net.IPNet, deviceNumber int, vpcCIDRs []string, useExternalSNAT bool,
-	netLink netlinkwrapper.NetLink, ns nswrapper.NS, mtu int, log logger.Logger, procSys procsyswrapper.ProcSys) error {
+// SetupPodNetwork wires up linux networking for a pod's network
+func (n *linuxNetwork) SetupPodNetwork(hostVethName string, contVethName string, netnsPath string, v4Addr *net.IPNet, v6Addr *net.IPNet,
+	deviceNumber int, mtu int, log logger.Logger) error {
+	log.Debugf("SetupPodNetwork: hostVethName=%s, contVethName=%s, netnsPath=%s, v4Addr=%v, v6Addr=%v, deviceNumber=%d, mtu=%d",
+		hostVethName, contVethName, netnsPath, v4Addr, v6Addr, deviceNumber, mtu)
 
-	hostVeth, err := setupVeth(hostVethName, contVethName, netnsPath, v4Addr, v6Addr, netLink, ns, mtu, procSys, log)
+	hostVeth, err := n.setupVeth(hostVethName, contVethName, netnsPath, v4Addr, v6Addr, mtu, log)
 	if err != nil {
-		return errors.Wrapf(err, "setupNS network: failed to setup veth pair.")
+		return errors.Wrapf(err, "SetupPodNetwork: failed to setup veth pair")
 	}
 
-	log.Debugf("Setup host route outgoing hostVeth, LinkIndex %d", hostVeth.Attrs().Index)
+	var containerAddr *net.IPNet
+	if v4Addr != nil {
+		containerAddr = v4Addr
+	} else if v6Addr != nil {
+		containerAddr = v6Addr
+	}
 
-	var addrHostAddr *net.IPNet
-	//We only support either v4 or v6 modes.
-	if v4Addr != nil && v4Addr.IP.To4() != nil {
-		addrHostAddr = &net.IPNet{
-			IP:   v4Addr.IP,
-			Mask: net.CIDRMask(32, 32)}
-	} else if v6Addr != nil && v6Addr.IP.To16() != nil {
-		addrHostAddr = &net.IPNet{
-			IP:   v6Addr.IP,
-			Mask: net.CIDRMask(128, 128)}
+	rtTable := unix.RT_TABLE_MAIN
+	if deviceNumber > 0 {
```

Comment:
Before we used to add the container rule only if device number is > 0 - https://github.com/aws/amazon-vpc-cni-k8s/blob/8855418090a8c69b0c8561113322fa1191cbe5da/cmd/routed-eni-cni-plugin/driver/driver.go#L321-L331 but now we will continue to setup the rule. Is it added on purpose?

Reply:
No, we will only add the "fromContainer" rule if the RTTable is not main.  (the check of deviceNumber > 0 is now replaced by rtTable != main inside the function been called )